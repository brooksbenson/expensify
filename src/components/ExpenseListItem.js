import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from './../actions/expenses';

function ExpenseListItem({ dispatch, createdAt, description, amount, id }) {
  return (
    <div>
      <h3>{description}</h3>
      <p>{amount} - {createdAt}</p>
      <button onClick={() => dispatch(removeExpense(id))}>x</button>
    </div>
  )
}

export default connect()(ExpenseListItem);  