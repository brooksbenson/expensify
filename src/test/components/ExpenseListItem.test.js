import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenseData from '../fixtures/expenses';

let [expense] = expenseData;
expense.createdAt = 0;

test('should render ExpenseListItem with expense data', () => {
  let wrapper = shallow(<ExpenseListItem {...expense} />);
  expect(wrapper).toMatchSnapshot();
});