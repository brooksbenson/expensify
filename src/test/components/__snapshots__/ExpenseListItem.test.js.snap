// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with expense data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      car wash
    </h3>
  </Link>
  <p>
    $10.00
    -
    December 31st, 1969
  </p>
</div>
`;
